src/jdk.jlink/share/classes/module-info.java

```diff
@@ -85,4 +85,6 @@
         jdk.tools.jlink.internal.plugins.VendorVersionPlugin,
         jdk.tools.jlink.internal.plugins.CDSPlugin,
         jdk.tools.jlink.internal.plugins.SaveJlinkArgfilesPlugin;
+
+    exports jdk.tools.jlink.internal to jdk.jpackage;
```

Comment:
I think we should try to have alternatives that avoid jpackage having a dependency on jlink internals.

Reply:
Yes, I feared this might be a concern. If we change how `jpackage` defines `ALL-DEFAULT` we could reduce it. The remaining use would be the case for issuing a warning for missing `jmods` directory, for which it's probably not worth keeping the dependency.

Input from `jpackage` devs would be appreciated.

Reply:
>  If we change how jpackage defines ALL-DEFAULT we could reduce it

I agree. jpackage can define ` ALL-DEFAULT` using public API, no need to use internal API.

Reply:
@alexeysemenyukoracle Do you mean it would be OK to re-define `ALL-DEFAULT` to not include `jdk.jlink` for runtime images? Would that be OK in general for `jpackage`?

Reply:
Exports from `jdk.jlink` to `jdk.jpackage` are not needed anymore.